dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/DHIS2BasicAuthenticationEntryPoint.java

```diff
@@ -28,43 +28,74 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+import com.google.common.base.MoreObjects;
 import org.apache.commons.lang.exception.ExceptionUtils;
 import org.hisp.dhis.dxf2.webmessage.WebMessageUtils;
 import org.hisp.dhis.render.RenderService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.MediaType;
 import org.springframework.security.authentication.LockedException;
 import org.springframework.security.core.AuthenticationException;
-import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint;
 
+import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
 /**
  * @author Viet Nguyen <viet@dhis2.org>
+ * @author Morten Svanæs <msvanaes@dhis2.org>
  */
-public class DHIS2BasicAuthenticationEntryPoint implements AuthenticationEntryPoint
+public class DHIS2BasicAuthenticationEntryPoint extends LoginUrlAuthenticationEntryPoint
 {
     @Autowired
     private RenderService renderService;
 
+    /**
+     * @param loginFormUrl URL where the login page can be found. Should either be
+     *                     relative to the web-app context path (include a leading {@code /}) or an absolute
+     *                     URL.
+     */
+    public DHIS2BasicAuthenticationEntryPoint( String loginFormUrl )
+    {
+        super( loginFormUrl );
+    }
+
     @Override
-    public void commence( HttpServletRequest request, HttpServletResponse response, AuthenticationException authException ) throws IOException
+    public void commence( HttpServletRequest request, HttpServletResponse response,
+        AuthenticationException authException )
+        throws IOException, ServletException
     {
-        String message;
+        String acceptHeader = MoreObjects.firstNonNull( request.getHeader( "Accept" ), "" );
```

Comment:
Can we use constants for the header names? You can find them in e.g. `com.google.common.net.HttpHeaders`.